sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -95,7 +93,123 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
 
   // Java api support
   def this(f: MapFunction[IN, java.lang.Double]) = this(x => f.call(x).asInstanceOf[Double])
+
   def toColumnJava: TypedColumn[IN, java.lang.Double] = {
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double)
+  extends Aggregator[IN, java.lang.Double, java.lang.Double] {
+
+  override def zero: java.lang.Double = null
```

Comment:
`TypedSum` is actually correct because it will return 0 in case of an empty set  ['In mathematics, an empty sum, or nullary sum, is a summation where the number of terms is zero. By convention,[1] the value of any empty sum of numbers is the additive identity, zero.'](https://en.wikipedia.org/wiki/Empty_sum). One could therefore argue that `Sum.scala` is  actually wrong because it returns null: `emptyTestData.agg(sum('key))`. We could either fix Sum.scala, although that might affect existing applications, or align both to return null, even though that is not technically correct.
The same does go for `TypedAvg`, which returns Double.Nan instead of null.